projects/packages/autoloader/src/class-latest-autoloader-guard.php

```diff
@@ -45,38 +45,31 @@ public function __construct( $plugins_handler, $autoloader_handler, $autoloader_
 	 * has the side-effect of actually loading the latest autoloader in the event that this
 	 * is not it.
 	 *
-	 * @param string   $current_plugin The current plugin we're checking.
-	 * @param string[] $plugins        The active plugins to check for autoloaders in.
+	 * @param string   $current_plugin             The current plugin we're checking.
+	 * @param string[] $plugins                    The active plugins to check for autoloaders in.
+	 * @param bool     $was_included_by_autoloader Indicates whether or not this autoloader was included by another.
 	 *
 	 * @return bool True if we should stop initialization, otherwise false.
 	 */
-	public function should_stop_init( $current_plugin, $plugins ) {
-		global $jetpack_autoloader_including_latest;
+	public function should_stop_init( $current_plugin, $plugins, $was_included_by_autoloader ) {
 		global $jetpack_autoloader_latest_version;
 
-		// When we're being included from an older autoloader we need to
-		// reset the latest version so that the new autoloader can look
-		// for the latest autoloader again.
-		if ( $jetpack_autoloader_including_latest ) {
-			$jetpack_autoloader_latest_version = null;
-		}
-
 		// We need to reset the autoloader when the plugins change because
 		// that means the autoloader was generated with a different list.
 		if ( $this->plugins_handler->have_plugins_changed( $plugins ) ) {
```

Comment:
I wanted to highlight the fact that the `Latest_Autoloader_Guard::should_stop_init()` method does not actually check to see if the current autoloader is the latest. We don't need to waste time loading the manifest to check because `Plugins_Handler::have_plugins_changed()` stores the `$all_plugins` list in a `$jetpack_autoloader_cached_plugin_paths` global and notes when it changes.

Since every version of the autoloader will try to load the latest we can always assume that we already have the latest if the global is unchanged. If there was a newer version of the plugin in this list it would have been picked up already by the last autoloader.